packages/nimble-components/src/table/models/table-validator.ts

```diff
@@ -72,27 +83,26 @@ export class TableValidator<TData extends TableNode> {
             return true;
         }
 
-        for (const record of data) {
-            if (
-                !Object.prototype.hasOwnProperty.call(
-                    record.clientRecord,
-                    idFieldName
-                )
-            ) {
-                this.missingRecordId = true;
-                continue;
+        if (typeof idFieldName === 'string') {
```

Comment:
I don't think this outer if-statement is providing any value. I should probably be removed.

Reply:
I went ahead and removed it.